hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -205,10 +211,11 @@ def print_end(self):
         By default does nothing.
         """
 
-    def print_step(self, step):
+    def print_step(self, step, result=None):
         """Print a step to the current reporter.
 
-        This is called right before a step is executed.
+        This is called right after a step is executed.
+        If the step execution raised an exception then "result" is None.
```

Comment:
The trouble with this is that `None` is also a valid value for the function to add to a Bundle - I'd suggest using Hypothesis' `not_set` sentinel value.

Reply:
I removed it, as requested